narwhals/_spark_like/namespace.py

```diff
@@ -67,6 +68,28 @@ def col(self, *column_names: str) -> SparkLikeExpr:
             *column_names, backend_version=self._backend_version, version=self._version
         )
 
+    def lit(self, value: object, dtype: DType | None) -> SparkLikeExpr:
+        if dtype is not None:
+            msg = "todo"
```

Comment:
We need `narwhals_to_native_dtype` for Spark.

Sitting on my laptop there is an oldish branch with `cast` that has `narwhals_to_native_dtype`.
Can this be a follow up?